radix.go

```diff
@@ -404,7 +417,18 @@ func parseRedisURL(urlStr string) (string, []DialOpt) {
 
 	var opts []DialOpt
 	q := u.Query()
-	if p, ok := u.User.Password(); ok {
+
+	n := u.User.Username()
```

Comment:
This whole section could probably be cleaned up into something like:

```go
const defaultAuthUser = "default" // also used in DialAuthPass

// ...

username := defaultAuthUser
if n := u.User.UserName(); n != "" {
    username = n
} else if n := q.Get("username"); n != "" {
    username = n
}

password := q.Get("password")
if p := u.User.Password(); p != nil {
    password = p
}

if password != "" {
    opts = append(opts, DialAuthUser(username, password))
}
```

Reply:
PTAL

I removed the condition around the `DialAuthUser` call since `Dial` already handles the default user and empty passwords